lib/reek/report.rb

```diff
@@ -0,0 +1,62 @@
+require_relative 'report/report'
+require_relative 'report/formatter'
+require_relative 'report/heading_formatter'
+
+module Reek
+  # Reek reporting functionality.
+  module Report
+    # Make report format symbol to a report class. Unknown symbols are mapped to +TextReport+.
+    #
+    # @param [Symbol] report_format The format to map
+    #
+    # @returns The mapped report class
+    #
+    # @api private
+    def self.report_class(report_format)
+      case report_format
+      when :yaml
+        YAMLReport
+      when :json
+        JSONReport
+      when :html
+        HTMLReport
+      when :xml
+        XMLReport
+      else # :text
+        TextReport
+      end
+    end
```

Comment:
Right, I thought about this as an option as well, but didn’t want to disrupt the current solution too much. But if so then let’s use `Hash#fetch`, that’s exactly its role:

```Ruby
REPORT_CLASSES = {
  html: HTMLReport,
  # ...
}

module_function

def report_class(report_format)
  REPORT_CLASSES.fetch(report_format) do
    fail "Unknown report format: #{report_format}"
  end
end
```